enos/modules/aws_iam_setup/main.tf

```diff
@@ -12,9 +12,9 @@ locals {
 }
 
 resource "aws_iam_user" "boundary" {
-  name                 = "boundary-e2e-${var.test_id}"
+  name                 = "demo-boundary-e2e-${var.test_id}"
```

Comment:
@kheina looks like it's [failing](https://github.com/hashicorp/boundary/actions/runs/7576191898/job/20634793940?pr=4255) now here with the error:

```
Error: creating IAM User (demo-boundary-e2e-JLsYY): NoSuchEntity: Scope ARN: arn:aws:iam::271311691044:policy/DemoUser does not exist or is not attachable.
	status code: 404, request id: fe36f99a-fff6-4874-9344-8486f2fd7cd8
```

Reply:
this repository uses a different aws account than the one these changes were tested with. that account does not have access to DemoUser yet, we are working to fix this

Reply:
cloudsec has given access to DemoUser and the enos tests are now passing 🎉